backend/models/upload.js

```diff
@@ -0,0 +1,301 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.ap
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+"use strict";
+
+const fs = require("fs");
+const multer = require("multer");
+const config = require("../config");
+const C = require("../constants");
+const middlewares = require("../middlewares/middlewares");
+const responseCodes = require("../response_codes");
+const systemLogger = require("../logger.js").systemLogger;
+const importQueue = require("../services/queue");
+const { modelStatusChanged } = require("./chatEvent");
+const { isValidTag } = require("./history");
+const { findModelSettingById, createCorrelationId } = require("./modelSetting");
+
+const handleChunkStream = async (req, filename) => {
+	return new Promise(resolve => {
+		const writeStream = fs.createWriteStream(filename, {encoding: "binary"});
+		req.on("data", (chunk) => writeStream.write(chunk, "binary"));
+		req.on("end", () => {
+			writeStream.end();
+			return resolve();
+		});
+	});
+};
+
+const stitchChunks = (corID, newFilename) => {
+	const sharedSpacePath = importQueue.getSharedSpacePath();
+	const filePath = sharedSpacePath + "/" + corID + "/chunks";
```

Comment:
I think `getSharedSpacePath` should be private for the queue - it's a bit high coupling otherwise.

Can we have a function in the queue service to return you a the path to corId? Something like
```suggestion
	const filePath = `${importQueue.getTaskFolder(corID)}/chunks`
```